### PR TITLE
route53_module_#2972

### PR DESCRIPTION
**Pull request by SteveLinden**
This routine is to create the logging for the route53 DNS that we have in place. To build the logging mechanism we need to do this per VPC. The module to pass through VPC details to set up logging. This routine will be called from the core-vpc section of this repo and will be an additional PR

The log build is in the monitoring section under core-logging

### CODE REVIEW

File: terraform/modules/route53-logging/variables.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+tags = {
```
Comment (dms1981) [inline terraform/modules/route53-logging/variables.tf:1]: Tags don't need to be supplied in a module, and won't work in this format. Consider having a variable that tags a map here instead.

File: terraform/modules/route53-logging/main.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+
+resource "aws_route53_resolver_query_log_config" "dns_logs" {
+  name            = "dns_logs"
+  destination_arn = aws_cloudwatch_log_group.aws_route53_logs_com.arn
+
+  tags = {
```
Comment (dms1981) [inline terraform/modules/route53-logging/main.tf:7]: `tags` map is empty and will need to be supplied with something. Perhaps a variable?

File: terraform/environments/core-sandbox/playground.tf [hunk 3]
```diff
@@ -19,16 +19,16 @@ module "ram-principal-association" {
...
-  #   values = ["oracle-linux-5.11*"]
-  # }
-
-}
-
-output "amis" {
-  value = data.aws_ami_ids.example.ids
-}
\ No newline at end of file
+# data "aws_ami_ids" "example" {
```
Comment (dms1981) [inline terraform/environments/core-sandbox/playground.tf:22]: Comments here should either be removed, or the regex should be fixed as this isn't relevant to the purpose of this PR

File: terraform/environments/core-logging/monitoring.tf [hunk 4]
```diff
@@ -2,3 +2,44 @@ module "core_monitoring" {
...
+resource "aws_cloudwatch_log_resource_policy" "route53-query-logging-policy" {
+  provider = aws.us-east-1
+
+  policy_document = data.aws_iam_policy_document.route53-query-logging-policy.json
+  policy_name     = "route53-query-logging-policy"
+}
+
+# Example Route53 zone with query logging
+
+resource "aws_route53_zone" "logs_com" {
```
Comment (dms1981) [inline terraform/environments/core-logging/monitoring.tf:43]: There's no need to create a new route53 zone is there?

File: terraform/environments/core-logging/monitoring.tf [hunk 5]
```diff
@@ -2,3 +2,44 @@ module "core_monitoring" {
...
+    resources = ["arn:aws:logs:*:*:log-group:/aws/route53/*"]
+
+    principals {
+      identifiers = ["route53.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_cloudwatch_log_resource_policy" "route53-query-logging-policy" {
```
Comment (dms1981) [inline terraform/environments/core-logging/monitoring.tf:34]: The provider here won't work

File: terraform/environments/core-logging/monitoring.tf [hunk 6]
```diff
@@ -2,3 +2,44 @@ module "core_monitoring" {
   source                     = "../../modules/core-monitoring"
   pagerduty_integration_keys = local.pagerduty_integration_keys
 }
+
+provider "aws" {
+  alias  = "eu-west-2"
+  region = "eu-west-2"
+}
+
+resource "aws_cloudwatch_log_group" "aws_route53_logs_com" {
```
Comment (dms1981) [inline terraform/environments/core-logging/monitoring.tf:11]: Does a provider need to be hardcoded? Isn't the default provider sufficient?

File: terraform/environments/core-logging/monitoring.tf [hunk 7]
```diff
@@ -2,3 +2,44 @@ module "core_monitoring" {
   source                     = "../../modules/core-monitoring"
   pagerduty_integration_keys = local.pagerduty_integration_keys
 }
+
+provider "aws" {
```
Comment (dms1981) [inline terraform/environments/core-logging/monitoring.tf:6]: What purpose does this provider fill? If it's needed it ought to be created in `providers.tf` for consistency / ease of comprehension

File: terraform/modules/route53-logging/main.tf (hunk 2 above)
Comment (SteveLinden) [reply terraform/modules/route53-logging/main.tf:7]: I was going to add the ones I put in the variable list. I'll have a think

File: terraform/modules/route53-logging/main.tf [hunk 8]
```diff
@@ -0,0 +1,15 @@
+
+
+resource "aws_route53_resolver_query_log_config" "dns_logs" {
+  name            = "dns_logs"
+  destination_arn = aws_cloudwatch_log_group.aws_route53_logs_com.arn
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/main.tf:5]: My plan is to build that first. Once it is in place I could then find the actual arn and use that. Not entirely sure that will work how I want but that was my thought process

File: terraform/environments/core-sandbox/playground.tf (hunk 3 above)
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf:22]: That's the section I commented out to get the plan to run. We need to look at it to see if the code can be amended to work correctly.

File: terraform/environments/core-logging/monitoring.tf (hunk 4 above)
Comment (SteveLinden) [reply terraform/environments/core-logging/monitoring.tf:43]: Forgot to remove this code

File: terraform/environments/core-logging/monitoring.tf (hunk 6 above)
Comment (SteveLinden) [reply terraform/environments/core-logging/monitoring.tf:11]: Copy and pasted the resource from terraform but forgot to remove the provider

File: terraform/environments/core-logging/monitoring.tf [hunk 9]
```diff
@@ -2,3 +2,29 @@ module "core_monitoring" {
   source                     = "../../modules/core-monitoring"
   pagerduty_integration_keys = local.pagerduty_integration_keys
 }
+resource "aws_cloudwatch_log_group" "aws_route53_logs_com" {
```
Comment (dms1981) [inline terraform/environments/core-logging/monitoring.tf:5]: Can you update the resource name here to something like `"route53-resolver-logs"`?

File: terraform/environments/core-sandbox/playground.tf [hunk 10]
```diff
@@ -19,16 +19,16 @@ module "ram-principal-association" {
 
 }
```
Comment (dms1981) [inline terraform/environments/core-sandbox/playground.tf:21]: As previously noted, this file doesn't need to be part of the commit. Can you remove the comments from the start of the lines here and raise an issue to fix this data call / take it on as a separate task?

File: terraform/modules/route53-logging/variables.tf (hunk 1 above)
Comment (SteveLinden) [reply terraform/modules/route53-logging/variables.tf]: Put in variables

File: terraform/modules/route53-logging/main.tf (hunk 8 above)
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/main.tf:5]: Amended to sort that

File: terraform/environments/core-logging/monitoring.tf (hunk 9 above)
Comment (SteveLinden) [reply terraform/environments/core-logging/monitoring.tf:5]: Amended to aws_route53_logs I think

File: terraform/environments/core-logging/providers.tf [hunk 11]
```diff
@@ -27,4 +27,8 @@ provider "aws" {
   assume_role {
     role_arn = "arn:aws:iam::${local.environment_management.account_ids["core-network-services-production"]}:role/ModernisationPlatformAccess"
   }
+}
+provider "aws" {
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/environments/core-logging/providers.tf:31]: Left for the time being. May be removable.

File: terraform/environments/core-sandbox/playground.tf (hunk 10 above)
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf:21]: Will do

File: terraform/modules/route53-logging/variables.tf [hunk 12]
```diff
@@ -0,0 +1,11 @@
+
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/variables.tf:1]: Amended

**Review by davidkelliott — CHANGES_REQUESTED** [review]
Hey, I've added a few additional comments, it would also be good in the PR description if you could explain in more detail what the PR is doing, what additional work is still needed etc.

File: terraform/environments/core-logging/monitoring.tf [hunk 13]
```diff
@@ -2,3 +2,29 @@ module "core_monitoring" {
...
+  name              = "aws_route53_zone.logs_com.name"
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/environments/core-logging/monitoring.tf]: Name changed but the plan was to move this out of this PR. I've created a new one which will contain, hopefully, ny other changes made to that.

File: terraform/environments/core-logging/providers.tf (hunk 11 above)
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/environments/core-logging/providers.tf:31]: Change made in another PR so it won't show in this one

File: terraform/modules/route53-logging/variables.tf (hunk 12 above)
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/variables.tf:1]: Commented out code removed

File: terraform/environments/core-sandbox/playground.tf (hunk 10 above)
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf:21]: Comments removed. Will raise an issue to fix this. Uses a very old version of linux so may no longer be available

File: terraform/environments/core-sandbox/playground.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf]: Removed the above from the PR so the original will stay in place.

File: terraform/environments/core-sandbox/playground.tf (hunk 10 above)
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf]: Removed the above from the PR so the original will stay in place.

File: terraform/environments/core-sandbox/playground.tf (hunk 3 above)
Comment (SteveLinden) [reply terraform/environments/core-sandbox/playground.tf]: Removed from the PR

File: terraform/modules/route53-logging/locals.tf [hunk 14]
```diff
@@ -0,0 +1,8 @@
+locals {
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/locals.tf]: The above changes have been made so this is resolved now

File: terraform/environments/core-logging/monitoring.tf (hunk 9 above)
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/environments/core-logging/monitoring.tf]: New routine (in the other PR) has been added and this have been moved

File: terraform/modules/route53-logging/main.tf [hunk 15]
```diff
@@ -0,0 +1,15 @@
+
+
+resource "aws_route53_resolver_query_log_config" "dns_logs" {
+  name            = "dns_logs"
+  destination_arn = "arn:aws:logs:eu-west-2:${local.environment_management.account_ids[terraform.workspace]}:log-group:aws_route53_logs:*"
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply terraform/modules/route53-logging/main.tf]: Changes made to this to pass in a variable which will be made available from the call in core-vpc (in another PR)

**Comment by SteveLinden** [comment]
The description was amended to add some more detail. I believe that this now explains the purpose in more detail.

**Pull request closed without merging**